pkg/kubelet/rkt/rkt.go

```diff
@@ -789,7 +789,12 @@ func (r *runtime) PullImage(image kubecontainer.ImageSpec, pullSecrets []api.Sec
 		tag = "latest"
 	}
 
-	creds, ok := r.dockerKeyring.Lookup(repoToPull)
+	keyring, err := credentialprovider.MakeDockerKeyring(pullSecrets, r.dockerKeyring)
```

Comment:
K, I figured as much.
On Tue, May 19, 2015 at 10:55 PM Yifan Gu <notifications@github.com> wrote:

> In pkg/kubelet/rkt/rkt.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7974#discussion_r30668138>
> :
>
> > @@ -789,7 +789,12 @@ func (r *runtime) PullImage(image kubecontainer.ImageSpec, pullSecrets []api.Sec
> >  		tag = "latest"
> >  	}
> >
> > -	creds, ok := r.dockerKeyring.Lookup(repoToPull)
> > +	keyring, err := credentialprovider.MakeDockerKeyring(pullSecrets, r.dockerKeyring)
>
> @lavalamp <https://github.com/lavalamp> @pmorie
> <https://github.com/pmorie> @deads2k <https://github.com/deads2k> This is
> because rkt can pull and run docker images with docker credentials, I read
> the keyring here and store it in the rkt's auth config files for docker (
> https://github.com/coreos/rkt/blob/master/Documentation/configuration.md#rktkind-dockerauth
> )
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7974/files#r30668138>
> .
>